src/Agent/NewRelic/Agent/Core/Logging/LoggerBootstrapper.cs

```diff
@@ -107,27 +108,38 @@ private static LoggerConfiguration ConfigureInMemoryLogSink(this LoggerConfigura
         }
 
         /// <summary>
-        /// Add the Event Log sink if running on .NET Framework
+        /// Add the Event Log sink if running on Windows
         /// </summary>
         /// <param name="loggerConfiguration"></param>
         private static LoggerConfiguration ConfigureEventLogSink(this LoggerConfiguration loggerConfiguration)
         {
-#if NETFRAMEWORK
-            const string eventLogName = "Application";
-            const string eventLogSourceName = "New Relic .NET Agent";
+#if NETSTANDARD2_0
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                const string eventLogName = "Application";
+                const string eventLogSourceName = "New Relic .NET Agent";
+                try
+                {
 
-            loggerConfiguration
-                    .WriteTo.Logger(configuration =>
-                    {
-                        configuration
-                        .ExcludeAuditLog()
-                        .WriteTo.EventLog(
-                            source: eventLogSourceName,
-                            logName: eventLogName,
-                            restrictedToMinimumLevel: LogEventLevel.Warning,
-                            outputTemplate: "{Level}: {Message}{NewLine}{Exception}"
-                        );
-                    });
+                    loggerConfiguration
+                        .WriteTo.Logger(configuration =>
+                        {
+                            configuration
+                                .ExcludeAuditLog()
+                                .WriteTo.EventLog(
+                                    source: eventLogSourceName,
+                                    logName: eventLogName,
+                                    restrictedToMinimumLevel: LogEventLevel.Warning,
+                                    outputTemplate: "{Level}: {Message}{NewLine}{Exception}",
+                                    manageEventSource: true
+                                );
+                        });
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e);
```

Comment:
I'm not sure automatically writing to the console is a good idea, even in an emergency, but I'm curious what others think.